repaygithub/cactus
# CACTUS-398 :: Button Polymorphism

Ticket: https://repayonline.atlassian.net/browse/CACTUS-398

Pretty straightforward...the only thing I want to mention is that the prop types for `as` are being `@ts-ignored` because apparently I'm not able to generate an array of keys from `JSX.IntrinsicElements`, so I don't have a way to explicitly state which strings are allowed without listing them all out by hand. Kind of annoying 😠

### Testing
1. Start the storybooks and open the new Button "as Link" story.
2. Inspect the html and make sure an `<a />` tag is rendered in place of `<button />`.
3. Make sure there aren't any style changes compared to a regular button
4. Click the button & make sure the link works. It should take you to Google.
